### PR TITLE
ReportsViewModel CurrentDateRange tests

### LINKED ISSUE #4991: Add more tests for ReportsViewModel.CurrentDateRangeStringObservable
While working on #4873 i noticed that `ReportsViewModel.CurrentDateRangeStringObservable` wasn't properly tested - it only had 1 test. Now I'm adding a few more tests for the bug that i'm fixing, but more tests are needed.

### PR DESCRIPTION
**Pull request by Odrakir**
## What's this?
Adds some more tests to `ReportsViewModel` `CurrentDateRange`

### Relationships
Closes #4991 

## Why do we want this?
Tests are good

## How is it done?
I'm using `MemeberData` magic to generate a bunch of values for testing.

## :squid: Permissions
Go!

### CODE REVIEW

File: Toggl.Core.Tests/UI/ViewModels/ReportsViewModelTests.cs [hunk 1]
```diff
@@ -229,37 +230,132 @@ public async Task IsInitializedToEmptyOrNull()
...
+                    var now = DateTimeOffset.Now;
+                    TimeService.CurrentDateTime.Returns(now);
+                    ViewModel.CurrentDateRange.Subscribe(observer);
+                    await ViewModel.Initialize();
+
+                    await ViewModel.LoadReport(0, startDate, endDate, ReportsSource.Calendar);
 
                     TestScheduler.Start();
-                    observer.LastEmittedValue().Should().Be($"{Resources.ThisWeek} ▾");
+                        observer.LastEmittedValue().Should().Be($"{result} ▾");
```
Comment (sofokli1) [inline Toggl.Core.Tests/UI/ViewModels/ReportsViewModelTests.cs]: why the extra indentation?

File: Toggl.iOS/ViewControllers/ReportsViewController.cs [hunk 2]
```diff
@@ -106,7 +106,8 @@ bool isWorkspaceNameTooLong(string workspaceName)
                 .Subscribe(WorkspaceLabel.Rx().Text())
                 .DisposedBy(DisposeBag);
 
-            ViewModel.CurrentDateRangeStringObservable
+            ViewModel.CurrentDateRange
+                .Debug("title")
```
Comment (sofokli1) [inline Toggl.iOS/ViewControllers/ReportsViewController.cs]: 🔥

File: Toggl.Core.Tests/UI/ViewModels/ReportsViewModelTests.cs [hunk 3]
```diff
@@ -229,37 +230,132 @@ public async Task IsInitializedToEmptyOrNull()
...
                 [Theory, LogIfTooSlow]
-                [InlineData(BeginningOfWeek.Monday)]
-                [InlineData(BeginningOfWeek.Tuesday)]
-                [InlineData(BeginningOfWeek.Wednesday)]
-                [InlineData(BeginningOfWeek.Thursday)]
-                [InlineData(BeginningOfWeek.Friday)]
-                [InlineData(BeginningOfWeek.Saturday)]
-                [InlineData(BeginningOfWeek.Sunday)]
-                public async Task EmitsThisWeekWhenCurrentWeekIsSelected(BeginningOfWeek beginningOfWeek)
+                [MemberData("allCombinations")]
```
Comment (arturs105) [inline Toggl.Core.Tests/UI/ViewModels/ReportsViewModelTests.cs]: You can use `nameof(allCombinations)` here and in the other places as well.

File: Toggl.Core.Tests/UI/ViewModels/ReportsViewModelTests.cs [hunk 4]
```diff
@@ -229,37 +230,132 @@ public async Task IsInitializedToEmptyOrNull()
...
+                }
+            }
+
+            public sealed class WhenARandomDateRangeIsSelected : ReportsViewModelTest
+            {
+                public static IEnumerable<object[]> randomCombinations
+                {
+                    get
+                    {
+                        yield return new object[] { new DateTime(2019, 5, 23), new DateTime(2019, 5, 29), "05/23 - 05/29" };
```
Comment (arturs105) [inline Toggl.Core.Tests/UI/ViewModels/ReportsViewModelTests.cs]: Testing this with different date formats wouldn't hurt.

Comment (Odrakir) [reply Toggl.Core.Tests/UI/ViewModels/ReportsViewModelTests.cs]: I thought about doing it, but we are not testing that really, I think it's an overkill

**Pull request merged**